python/pyspark/sql/connect/client.py

```diff
@@ -409,7 +409,7 @@ def __init__(
         self._builder = ChannelBuilder(connectionString, channelOptions)
         self._user_id = None
         self._retry_policy = {
-            "max_retries": 15,
+            "max_retries": 4,
```

Comment:
I don't mind about this number but just curious. Is there a reason to pick 4?

Reply:
See PR description.